examples/grpc/greeter/client.cpp

```diff
@@ -0,0 +1,84 @@
+#include "greeter.grpc.fb.h"
+#include "greeter_generated.h"
+
+#include <grpc++/grpc++.h>
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+class GreeterClient {
+ public:
+  GreeterClient(std::shared_ptr<grpc::Channel> channel)
+    : stub_(Greeter::NewStub(channel)) {}
+
+  std::string SayHello(const std::string& name) {
+    auto name_offset = mb_.CreateString(name);
+    auto request_offset = CreateHelloRequest(mb_, name_offset);
+    mb_.Finish(request_offset);
+    auto request_msg = mb_.ReleaseMessage<HelloRequest>();
+
+    flatbuffers::grpc::Message<HelloReply> response_msg;
+
+    grpc::ClientContext context;
+
+    auto status = stub_->SayHello(&context, request_msg, &response_msg);
+    if (status.ok()) {
+      const HelloReply* response = response_msg.GetRoot();
+      return response->message()->str();
+    } else {
+      std::cerr << status.error_code() << ": " << status.error_message()
+                << std::endl;
+      return "RPC failed";
+    }
+  }
+
+  template <class F>
```

Comment:
It's not a big deal either way, there's certainly some inconsistencies in style due to external contributors, things I didn't catch.

No idea how to express that in clang-format. There's quite a few things the Google C++ Style Guide leaves unspecified, so it defaults to "how the project has always done it".

I'm a big fan of clang-format, but running it will typically polute your commit with things from older changes. Unless we enforced running it on merge somehow it's hard to rely on it. It also doesn't support the pre-processor indentation and a few other things. Usefully, some editor integrations allow you to clang-format just a selection.